lib/galaxy/dependencies/pipfiles/default/pinned-dev-requirements.txt

```diff
@@ -43,6 +43,7 @@ pytest-metadata==1.8.0
 pytest-postgresql==1.4.1
 pytest-pythonpath==0.7.3
 pytest==4.6.9
+python-irodsclient==0.8.2
```

Comment:
I think this should be added to lib/galaxy/dependencies/conditional-requirements.txt (there's a note about PyRods already).

Reply:
Thx, will move it there.

Reply:
The file has this comment "These dependencies are only required when certain config options are set".  On a second thought, should irods be moved there or stay?

Reply:
If we have tests that utilize irods (yes) I think we should have it in both places.
`pinned-dev-requirements.txt` dependencies will be installed when running tests, the stuff in `conditional-requirements.txt` will be installed based on config values (in normal operation). The conditional dependency check doesn't work for dynamically constructed test config settings.

Reply:
For conditional requirements have a look at the pythos object store and its kamaki package dependency at https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/dependencies/__init__.py#L175

Reply:
Indeed, that is how you do it, but we still need it in lib/galaxy/dependencies/pipfiles/default/pinned-dev-requirements.txt

Reply:
Ok, added it to conditional-requirements.txt  as well.

Reply:
@kxk302 Can you also add `python-irodsclient = "*"` to the `[dev-packages]` section of `lib/galaxy/dependencies/pipfiles/default/Pipfile` and run `make update-dependencies` ?

This should add the required `xmlrunner` to `lib/galaxy/dependencies/pipfiles/default/pinned-dev-requirements.txt` (though there may be unrelated changes to the `*requirements.txt` files you can ignore.

Reply:
@nsoranzo Added the line and ran the command. It changed many versions in the following files:

lib/galaxy/dependencies/pipfiles/default/pinned-dev-requirements.txt
lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt
lib/galaxy/dependencies/pipfiles/flake8/pinned-requirements.txt

Not sure if these changes are valid (or have side effects)

Reply:
You can use `git add -u -p lib/galaxy/dependencies/pipfiles/` to select only the relevant changes (xmlrunner line added to `lib/galaxy/dependencies/pipfiles/default/pinned-dev-requirements.txt` , I think), then `git checkout -- lib/galaxy/dependencies/pipfiles/` to get rid of the unwanted changes.

Reply:
Thanks. I got rid of un-wanted changes.

Reply:
prettytable and xmlrunner should also have been added here, will be fixed in https://github.com/galaxyproject/galaxy/pull/9529